pkg/snowflake/parameters.go

```diff
@@ -779,16 +864,35 @@ func ShowObjectParameter(db *sql.DB, key string, objectType ObjectType, objectId
 	return &value, nil
 }
 
-func ListParameters(db *sql.DB, parameterType ParameterType, pattern string) ([]Parameter, error) {
+func ListAccountParameters(db *sql.DB, pattern string) ([]Parameter, error) {
 	var stmt string
-	if parameterType == ParameterTypeAccount || parameterType == ParameterTypeSession {
-		if pattern != "" {
-			stmt = fmt.Sprintf("SHOW PARAMETERS LIKE '%s' IN %v", pattern, parameterType)
-		} else {
-			stmt = fmt.Sprintf("SHOW PARAMETERS IN %v", parameterType)
+	if pattern != "" {
+		stmt = fmt.Sprintf("SHOW PARAMETERS LIKE '%s' IN ACCOUNT", pattern)
+	} else {
+		stmt = "SHOW PARAMETERS IN ACCOUNT"
+	}
+	log.Printf("[DEBUG] query = %s", stmt)
+	rows, err := db.Query(stmt)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	params := []Parameter{}
+	if err := sqlx.StructScan(rows, &params); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, nil
 		}
+		return nil, fmt.Errorf("unable to scan row for %s err = %w", stmt, err)
+	}
+	return params, nil
+}
+
+func ListSessionParameters(db *sql.DB, pattern string, user string) ([]Parameter, error) {
```

Comment:
Lots of code duplication between `ListAccountParameters` and `ListSessionParameters`, can we factorize?